word_search.py

```diff
@@ -0,0 +1,102 @@
+        self.__words_to_search.put([grid[column][row], column, row, []])
+        
+        while not self.__words_to_search.empty():
+            currentArray = self.__words_to_search.get()
+            currentWord = currentArray[0]
+            column = currentArray[1]
+            row = currentArray[2]
+            positions = currentArray[3]
+            
+            if dictionary.isPrefix(currentWord):
+                tempPositions = positions.copy()
+                tempPositions.append([column, row])
+                if tempPositions in self.__pathsTaken: continue
+                
+                if dictionary.isWord(currentWord) and currentWord not in self.__foundWords:
+                    self.__foundWords.append(currentWord)
+                    newPositions = positions.copy()
+                    newPositions.append([column, row])
+                    self.__pathsTaken.append(newPositions)
+            
+                if row+1 < rowNo and [column, row+1] not in positions:
+                    self.addWordToSearch(currentWord, column, row, column, row+1, positions, grid, dictionary)
+                    
+                if row-1 > -1 and [column, row-1] not in positions:
+                    self.addWordToSearch(currentWord, column, row, column, row-1, positions, grid, dictionary)
+                    
+                if column+1 < columnNo and [column+1, row] not in positions:
+                    self.addWordToSearch(currentWord, column, row, column+1, row, positions, grid, dictionary)
+                
+                if column-1 > -1 and [column-1, row] not in positions:
+                    self.addWordToSearch(currentWord, column, row, column-1, row, positions, grid, dictionary)
+                
+                if column-1 > -1 and row-1 > -1 and [column-1, row-1] not in positions:
+                    self.addWordToSearch(currentWord, column, row, column-1, row-1, positions, grid, dictionary)
+                
+                if column+1 < columnNo and row+1 < rowNo and [column+1, row+1] not in positions:
+                    self.addWordToSearch(currentWord, column, row, column+1, row+1, positions, grid, dictionary)
+                
+                if column-1 > -1 and row+1 < rowNo and [column-1, row+1] not in positions:
+                    self.addWordToSearch(currentWord, column, row, column-1, row+1, positions, grid, dictionary)
+            
+                if column+1 < columnNo and row-1 > -1 and [column+1, row-1] not in positions:
+                    self.addWordToSearch(currentWord, column, row, column+1, row-1, positions, grid, dictionary)
+            
+            else:
+                previousPathTaken = positions.copy()
+                previousPathTaken.append([column, row])
+                self.__pathsTaken.append(previousPathTaken)
+                
+                if not self.__words_to_search.empty(): continue
+                
+                rowAddCopy = previousPathTaken.copy()
```

Comment:
This is taken care of the fact that you might need to start from different starting position.
There are, however, better ways to deal with it.
If you think of it, right now you put (0,0) as the starting point in the queue, but, for this problem, you have multiple starting points (any cell can be the starting point), so you should encode that instead of trying to do this here.